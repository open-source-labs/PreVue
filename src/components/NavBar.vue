<template>
  <!--the navbar on the main and tree views-->
  <v-app-bar
    class="border-bottom"
    id="navbar"
    color="black"
    role="navigation"
    aria-label="main navigation"
  >
    <img src="../assets/logo.png" id="prevue-logo" @click="routeHome" />
    <h1 id="prevue">PreVue</h1>
    <div class="buttons">
      <v-btn id="tree-btn" @click="routeTree">
        <br />
        <span class="white--text">Tree</span>
      </v-btn>

      <SaveProjectComponent v-if="validUser"></SaveProjectComponent>
      <ExportProjectComponent></ExportProjectComponent>
      <OpenProjectComponent
        :key="rerenderKey"
        v-if="validUser"
      ></OpenProjectComponent>
      <LogOutComponent v-if="validUser"></LogOutComponent>
    </div>
  </v-app-bar>
</template>

<script>
import SaveProjectComponent from '@/components/SaveProjectComponent.vue';
import OpenProjectComponent from '@/components/OpenProjectComponent.vue';
import ExportProjectComponent from '@/components/ExportProjectComponent.vue';
import LogOutComponent from './LogOutComponent.vue';

export default {
  name: 'NavBar',
  props: ['route', 'imageURL'],

  components: {
    SaveProjectComponent,
    OpenProjectComponent,
    LogOutComponent,
    ExportProjectComponent
  },
  methods: {
    // routing to homepage and tree view
    routeHome() {
      this.$router.push({ path: '/home' });
    },
    routeTree() {
      this.$router.push({ path: '/tree' });
    }
  },
  computed: {
    // checks if user is loggedin and conditonally renders save and open project buttons
    validUser() {
      return this.$store.state.loggedIn;
    },
    // uforces update of list of projects upon saving of new project
    rerenderKey: {
      get() {
        return this.$store.state.rerenderKey;
      }
    }
  }
};
</script>

<style scoped>
@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@300&display=swap');
.border-bottom {
  border-bottom: 1px solid #ccc;
  border-color: #39b982;
  align-items: right;
}
.buttons {
  width: 100vw;
  padding-right: 1em;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  vertical-align: middle;
}

#prevue {
  font-family: 'Nunito', sans-serif;
  font-weight: 700;
  font-size: 2.5em;
  color: #39b982;
  text-decoration: none;
  margin-right: 10px;
}
#prevue-logo {
  height: 80px;
  width: 70px;
  transform: scale(0.7);
}

#prevue-logo:hover {
  cursor: pointer;
}

#tree-btn {
  font-weight: 700;
  font-size: 14px;
}

button {
  margin: 5px;
  font-size: 14px;
  background: none;
  border: none;
}
button:hover {
  color: #00c4a7;
}
</style>
