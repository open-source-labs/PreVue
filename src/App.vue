<template>
  <v-card id="app">
    <v-layout>
      <NavBar
        v-if="!$route.meta.hideNavbar"
        id="nav"
        imageUrl="../src/assets/logo.png"
      ></NavBar>

      <v-main style="min-height: 300px"><router-view /></v-main>
    </v-layout>
  </v-card>
</template>

<script>
import NavBar from '@/components/NavBar.vue';

export default {
  components: {
    NavBar,
  },
};
</script>

<style lang="scss">
@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@300&display=swap');
#app {
  height: 105vh;
  font-family: 'Nunito', sans-serif;
}
#nav {
  display: flex;
  justify-content: space-between;
}
</style>
