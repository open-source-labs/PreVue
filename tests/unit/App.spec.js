import { shallowMount, mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import { createRouter, createWebHistory } from 'vue-router';
import { createStore } from 'vuex';
import App from '@/App.vue';
import NavBar from '@/components/NavBar.vue';

describe('App.vue', () => {
  const router = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: '/',
        component: App,
        meta: {
          hideNavbar: true
        }
      }
    ]
  });

  const store = createStore({
    state() {
      return {};
    }
  });

  it('contains v-main element that wraps router-view', () => {
    const wrapper = shallowMount(App, {
      global: {
        plugins: [router, store]
      }
    });
    expect(wrapper.find('v-main')).toBeTruthy();
    expect(wrapper.find('v-main > router-view')).toBeTruthy();
  });
});

describe('App.vue', () => {
  // create store with fake properties
  const store = createStore({
    state() {
      return {
        store1: '',
        store2: '',
        componentNameInputValue: ''
      };
    }
  });

  const router = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: '/',
        component: App,
        meta: {
          hideNavbar: true
        }
      }
    ]
  });
  it('should renders elements found in children', () => {
    const wrapper = mount(App, {
      components: {
        NavBar
      },
      global: {
        plugins: [store, router]
      }
    });
    expect(wrapper.find('h1').exists()).toBeTruthy();
  });

  it('should renders elements found in children', () => {
    const wrapper = mount(App, {
      components: {
        NavBar
      },
      global: {
        plugins: [store, router]
      }
    });
    expect(wrapper.findComponent({ name: 'NavBar' }).exists()).toBeTruthy();
  });
});
