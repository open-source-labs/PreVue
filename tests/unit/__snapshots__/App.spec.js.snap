// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > renders App correctly 1`] = `
<div
  data-v-app=""
>
  
  <v-card
    id="app"
  >
    <v-layout>
      <nav-bar-stub
        id="nav"
        imageurl="../src/assets/logo.png"
      />
      <v-main
        style="min-height: 300px;"
      >
        <router-view />
      </v-main>
    </v-layout>
  </v-card>
  <!-- &lt;v-app&gt;
    &lt;div id="app" class="hero"&gt;
      &lt;NavBar v-if="!$route.meta.hideNavbar"  id="nav" imageUrl="../src/assets/logo.png"&gt;&lt;/NavBar&gt;

      &lt;section&gt;
        &lt;router-view /&gt;
      &lt;/section&gt;
    &lt;/div&gt;
  &lt;/v-app&gt; -->
  
</div>
`;

exports[`App.vue > renders App correctly 2`] = `
<div
  data-v-app=""
>
  
  <v-card
    id="app"
  >
    <v-layout>
      <nav-bar-stub
        id="nav"
        imageurl="../src/assets/logo.png"
      />
      <v-main
        style="min-height: 300px;"
      >
        <router-view />
      </v-main>
    </v-layout>
  </v-card>
  <!-- &lt;v-app&gt;
    &lt;div id="app" class="hero"&gt;
      &lt;NavBar v-if="!$route.meta.hideNavbar"  id="nav" imageUrl="../src/assets/logo.png"&gt;&lt;/NavBar&gt;

      &lt;section&gt;
        &lt;router-view /&gt;
      &lt;/section&gt;
    &lt;/div&gt;
  &lt;/v-app&gt; -->
  
</div>
`;

exports[`App.vue > renders html correct 1`] = `
<div
  data-v-app=""
>
  
  <v-card
    id="app"
  >
    <v-layout>
      <nav-bar-stub
        id="nav"
        imageurl="../src/assets/logo.png"
      />
      <v-main
        style="min-height: 300px;"
      >
        <router-view />
      </v-main>
    </v-layout>
  </v-card>
  <!-- &lt;v-app&gt;
    &lt;div id="app" class="hero"&gt;
      &lt;NavBar v-if="!$route.meta.hideNavbar"  id="nav" imageUrl="../src/assets/logo.png"&gt;&lt;/NavBar&gt;

      &lt;section&gt;
        &lt;router-view /&gt;
      &lt;/section&gt;
    &lt;/div&gt;
  &lt;/v-app&gt; -->
  
</div>
`;

exports[`App.vue > renders html correctly 1`] = `
<div
  data-v-app=""
>
  
  <v-card
    id="app"
  >
    <v-layout>
      <nav-bar-stub
        id="nav"
        imageurl="../src/assets/logo.png"
      />
      <v-main
        style="min-height: 300px;"
      >
        <router-view />
      </v-main>
    </v-layout>
  </v-card>
  <!-- &lt;v-app&gt;
    &lt;div id="app" class="hero"&gt;
      &lt;NavBar v-if="!$route.meta.hideNavbar"  id="nav" imageUrl="../src/assets/logo.png"&gt;&lt;/NavBar&gt;

      &lt;section&gt;
        &lt;router-view /&gt;
      &lt;/section&gt;
    &lt;/div&gt;
  &lt;/v-app&gt; -->
  
</div>
`;

exports[`App.vue renders correct 1`] = `
<div
  class="hero is-fullHeight"
  id="app"
>
  <navbar-stub />
   
  <section
    class="section"
  >
    <router-view />
  </section>
</div>
`;
